framework/src/Volo.Abp.AspNetCore.MultiTenancy/Volo/Abp/AspNetCore/MultiTenancy/FormTenantResolveContributor.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Volo.Abp.MultiTenancy;
+
+namespace Volo.Abp.AspNetCore.MultiTenancy
+{
+    public class FormTenantResolveContributor : HttpTenantResolveContributorBase
+    {
+        public const string ContributorName = "Form";
+
+        public override string Name => ContributorName;
+
+        protected override string GetTenantIdOrNameFromHttpContextOrNull(ITenantResolveContext context, HttpContext httpContext)
+        {
+            if (httpContext.Request == null || !httpContext.Request.Form.Any())
+            {
+                return null;
+            }
+
+            return httpContext.Request.Form[context.GetAbpAspNetCoreMultiTenancyOptions().TenantKey];
```

Comment:
Use the async method to get value from the form will be better.
Maybe we should make `ITenantResolver` support `async`.
```
var form = await HttpRequest.ReadAsFormAsync();
return form["context.GetAbpAspNetCoreMultiTenancyOptions().TenantKey"];
```

Reply:
I agree. May resolve tenants from `JSON`, asynchronous is better. But this is a breaking change. @hikalkan  What do you think?

Reply:
We don't want to introduce such a breaking change now.